user_gen.py

```diff
@@ -0,0 +1,178 @@
+
+    def rand_num_user_gen(self, params=None, func=None):
+        """
+        generates an integer number of users
+        params is a dict of parameters
+        func is a function with a signature like numpy.random functions
+        """
+        if params is None:
+            params = {"lam": 5,
+                      "plus": 1
+                      }
+        if func is None:
+            func = poisson_plus
+        self.users_per_job = func(**params, size=self.num_jobs)
+
+    def rand_start_times_gen(self, params=None, func=None):
+        """
+        params is a dict of parameters
+        func is a function with a signature like numpy.random functions
+        """
+        if params is None:
+            params = {"rate": 5}
+        if func is None:
+            func = simp_poisson_process
+        self.start_times = func(**params, size=self.num_jobs)
+
+    def make_job_set(self, job_type=None):
+        if job_type is None:
+            job_type = Simple_Job
+        job_array = list(np.zeros(shape=self.num_jobs))
+        for i, x in enumerate(job_array):
+            job_array[i] = job_type(self.users_per_job[i])
+        self.jobs = job_array
+        return self.jobs
+
+
+class Simple_Job(object):
+    def __init__(self, num_users):
+        self.num_users = num_users
+
+    @property
+    def wait_times(self):
+        return self._wait_times
+
+    @wait_times.setter
+    def wait_times(self, wait_times):
+        if all([x >= 0 for x in wait_times]):
+            self._wait_times = wait_times
+        else:
+            raise ValueError("Wait times must be positive")
+
+    @property
+    def end_times(self):
+        return self._end_times
+
+    @end_times.setter
+    def end_times(self, end_times):
+        if all([x > 0 for x in end_times]):
+            self._end_times = end_times
+        else:
+            raise ValueError("End times must be positive")
+
+    def rand_wait_times_gen(self, params=None, func=None):
+        if params is None:
+            params = {"rate": 5}
+        if func is None:
+            func = diffs_btw_exp
+        self.wait_times = func(**params, size=self.num_users)
+
+    def rand_end_times_gen(self, params=None, func=None):
+        if params is None:
+            params = {"scale": 5}
+        if func is None:
+            func = np.random.exponential
+        self.wait_times = func(**params, size=self.num_users)
+
+
+class JHub_Job_Set(Simple_Job_Set):
+
+    def rand_num_user_gen(self, mean=1):
+        params = {"lam": mean,
+                  "plus": 1
+                  }
+        func = poisson_plus
+        super(JHub_Job_Set, self).rand_num_user_gen(params=params, func=func)
+
+    def rand_start_times_gen(self, mean=5):
+        params = {"rate": mean}
+        func = simp_poisson_process
+        super(JHub_Job_Set, self).rand_start_times_gen(params=params, func=func)
+
+    def make_job_set(self):
```

Comment:
That probably won't be necessary its a instance attribute since the code can just be self-referential and that should be fine.